common/reconciliation/fetcher/timer_test.go

```diff
@@ -0,0 +1,177 @@
+					Return(&persistence.GetTimerIndexTasksResponse{
+						Timers:        timerTasks,
+						NextPageToken: nil,
+					}, nil)
+
+				mockRetryer.EXPECT().GetShardID().Return(123)
+
+				return timerTasks
+			},
+			expectedPage: pagination.Page{
+				Entities: []pagination.Entity{
+					&entity.Timer{
+						ShardID:             123,
+						DomainID:            "testDomainID",
+						WorkflowID:          "testWorkflowID",
+						RunID:               "testRunID",
+						TaskType:            persistence.TaskTypeUserTimer,
+						VisibilityTimestamp: fixedTimestamp,
+					},
+				},
+			},
+			expectedError: false,
+		},
+		{
+			name: "Non-nil Pagination Token Passed",
+			setupMock: func() []*persistence.TimerTaskInfo {
+				nonNilToken := []byte("non-nil-token")
+
+				mockRetryer.EXPECT().
+					GetTimerIndexTasks(ctx, &persistence.GetTimerIndexTasksRequest{
+						MinTimestamp:  minTimestamp,
+						MaxTimestamp:  maxTimestamp,
+						BatchSize:     pageSize,
+						NextPageToken: nonNilToken,
+					}).
+					Return(&persistence.GetTimerIndexTasksResponse{
+						Timers:        nil, // Adjust based on expected behavior
+						NextPageToken: nonNilToken,
+					}, nil)
+
+				return nil // No timers expected in response
+			},
+			expectedPage: pagination.Page{
+				Entities:     nil,
+				CurrentToken: nonNilToken,
+				NextToken:    nonNilToken,
+			},
+			expectedError: false,
+		},
+		{
+			name: "Invalid Timer Causes Error",
+			setupMock: func() []*persistence.TimerTaskInfo {
+				// Example setup of an invalid timer
+				invalidTimer := &persistence.TimerTaskInfo{
+					DomainID:            "",               // Invalid as it's empty
+					WorkflowID:          "testWorkflowID", // Valid
+					RunID:               "testRunID",      // Valid
+					VisibilityTimestamp: time.Time{},      // Invalid as it's the zero value
+					TaskType:            persistence.TaskTypeUserTimer,
+				}
+
+				mockRetryer.EXPECT().
+					GetTimerIndexTasks(ctx, &persistence.GetTimerIndexTasksRequest{
+						MinTimestamp:  minTimestamp,
+						MaxTimestamp:  maxTimestamp,
+						BatchSize:     pageSize,
+						NextPageToken: nil,
+					}).
+					Return(&persistence.GetTimerIndexTasksResponse{
+						Timers:        []*persistence.TimerTaskInfo{invalidTimer},
+						NextPageToken: nil,
+					}, nil)
+
+				mockRetryer.EXPECT().GetShardID().Return(123)
+
+				return []*persistence.TimerTaskInfo{invalidTimer}
+			},
+			expectedPage:  pagination.Page{}, // No page is expected due to error
+			expectedError: true,              // An error is expected due to the invalid timer
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			tc.setupMock()
+
+			var token pagination.PageToken
+			if tc.name == "Non-nil Pagination Token Passed" {
+				token = nonNilToken
+			}
+			fetchFn := getUserTimers(mockRetryer, minTimestamp, maxTimestamp, pageSize)
+			page, err := fetchFn(ctx, token)
+
+			require.Equal(t, tc.expectedError, err != nil)
```

Comment:
it is a bit hard to read - I personally have to wrap my head around every time I see this 🤯 

I know, you have to do this to avoid if-s in table-tests, but maybe you can check for specific error type instead of expected or not? Then it will be natural to just check error against `expectedError error` field of the table-test.

Reply:
The code doesn't seem to have any custom error struct but I will give this a shot.

Reply:
okay! I tried whatever I could.

Thanks for the suggestion.